job-service-contract/src/main/resources/com/hpe/caf/services/job/swagger.yaml

```diff
@@ -415,6 +417,58 @@ definitions:
         example:
           "tag:4": "4"
           owner: "bob"
+  expiry:
+    type: object
+    properties:
+      "Active":
+        $ref: "#/definitions/policy"
+      "Completed":
+        $ref: "#/definitions/policy"
+      "Failed":
+        $ref: "#/definitions/policy"
+      "Cancelled":
+        $ref: "#/definitions/policy"
+      "Waiting":
+        $ref: "#/definitions/policy"
+      "Paused":
+        $ref: "#/definitions/policy"
+      "Expired":
+        $ref: "#/definitions/deletePolicy"
+      "Default":
+        $ref: "#/definitions/policy"
+    description: The expiration policy to be applied on the job
+  deletePolicy:
+    type: object
+    properties:
+      expiryTime:
+        $ref: "#/definitions/expiryTimePattern"
+      operation:
+        type: string
+        enum:
+          - delete
+    description: The expiration details to be applied on the job
+  expiryTimePattern:
+    type: object
```

Comment:
Why turn this into an object?

Reply:
If I pass it as a String to a new definition, it becomes an object and I get the following on compilation:

> incompatible types: java.lang.String cannot be converted to com.hpe.caf.services.job.client.model.ExpiryTimeDefinition

I couldn't manage to cast it either. So from that, I think that we can either create a proper object or duplicate the definition.
Found a report of that issue [here](https://github.com/swagger-api/swagger-parser/issues/243) or [here](https://github.com/swagger-api/swagger-codegen/issues/2314)